stdlib/public/core/Codable.swift

```diff
@@ -2894,13 +2894,28 @@ public extension RawRepresentable where RawValue == String, Self : Decodable {
 //===----------------------------------------------------------------------===//
 
 // FIXME: Uncomment when conditional conformance is available.
-extension Array : Codable /* where Element : Codable */ {
-    public init(from decoder: Decoder) throws {
-        guard Element.self is Decodable.Type else {
-            preconditionFailure("Array<\(Element.self)> does not conform to Decodable because \(Element.self) does not conform to Decodable.")
+extension Array : Encodable /* where Element : Encodable */ {
```

Comment:
Can we review the source compatibility consequences of making this change once we have conditional conformance? I want to be certain we have a good understanding for what user code will need to change once this is introduced. For example, generic algorithms will need additional `where` clauses, we will need overloads that still throw available in compatibility mode.